ui/src/pages/Login/AuthorizedCallback.tsx

```diff
@@ -13,21 +19,20 @@ export const AuthorizedCallback: React.FC = () => {
       </>)
     }
 
-    localStorage.setItem("token", token)
+    const rawUserData = jwtDecode(token) as AssumedTokenType;
 
-    getUserInfo().then(async res => {
-        // TODO: Actually cast data, instead of capture everything
-        const userInfo = await res.json()
-        localStorage.setItem("userData", JSON.stringify(userInfo))
+    // .sub and .aud shouldn't be long-lived, as this is a misuse of claims
+    // Expect these to change before long
+    const userInfo: UserInfo = {
+      avatar:   rawUserData.aud,
+      username: rawUserData.sub
+    };
 
-        // Redirect to homepage, we don't need to stay here!
-        return window.location.replace("/register");
-    })
+    localStorage.setItem("token", token)
+    localStorage.setItem("userData", JSON.stringify(userInfo))
```

Comment:
Since userData can be derived from the token, I personally wouldn't bother persisting it.

Reply:
Agreed, we can remove this later on; this is just what was there before 😅